src/ir_Kelvinator.cpp

```diff
@@ -135,7 +138,9 @@ void IRsend::sendKelvinator(unsigned char data[], uint16_t nbytes,
     space(KELVINATOR_GAP_SPACE * 2);
   }
 }
+#endif  // SEND_KELVINATOR
 
+#if (SEND_KELVINATOR || DECODE_KELVINATOR)
```

Comment:
Not that it really matters here, but for many of these (eg: `SEND_KELVINATOR`) the compiler will just not include anything that's not called - which works great for sending (so all A/C send codes could be included all the time). For decoding it's good to have a compile time option as it'll always get called if enabled.

Just a thought for other code as it's something I have noticed a couple of times.

Reply:
I concur. However, for most cases, decode() [the main one, not the sub-ones like decodeKelvinator()] includes the per-protocol decode routines, and most of the new A/C decoding routines pull in the class via the corresponding validChecksum() routines. They do so as a static function, but I don't know if the linker/compiler is smart enough to not pull in the whole object  or not. Thus, \<cough>this way\</cough> makes sure.

The SEND_BLAH & DECODE_BLAHs are really there so people can edit out easily what they don't need decoded to make it faster & smaller, if they are on a tight resource budget.
The _upstream_ Arduino IRremote library moved to doing that, and when I refactored most of the code, I decided to try to follow their spirit etc.

FYI, there is one exception to the sending bit. For some (legacy) reason, there is a [send()](https://github.com/markszabo/IRremoteESP8266/blob/master/src/IRsend.cpp#L275) function, which contains most of the sendBLAH() routines. I think most people probably don't use it, but if they do, they'll get the whole load of them.

If I was to place a bet, I'd wager that no-one bothers turning off protocols using this method, but .. it doesn't cost much to maintain.

I know, I know, I can't protect the users from themselves, but I try to give them a rifle that aims away from their feet when ever I can.